src/session/query.c

```diff
@@ -213,19 +219,12 @@ int _z_trigger_query_reply_final(_z_session_t *zn, const _z_reply_context_t *rep
 
     // The reply is the final one, apply consolidation if needed
     if (pen_qry->_consolidation == Z_CONSOLIDATION_MODE_LATEST) {
-        _z_pending_reply_list_t *pen_rps = pen_qry->_pending_replies;
-        _z_pending_reply_t *pen_rep = NULL;
-        while (pen_rps != NULL) {
-            pen_rep = _z_pending_reply_list_head(pen_rps);
+        while (pen_qry->_pending_replies != NULL) {
+            _z_pending_reply_t *pen_rep = _z_pending_reply_list_head(pen_qry->_pending_replies);
 
-            // Check if this is the same resource key
             // Trigger the query handler
-            if (_z_keyexpr_intersect(pen_qry->_key._suffix, strlen(pen_qry->_key._suffix),
```

Comment:
I believe this check shouldn't be removed to guarantee that all the replies are for the query being issued.

Reply:
As discussed internally, agreed

Reply:
I agree. We must ensure that the user do not get a reply outside the requested set of keyexpr.